### ISSUE TITLE
Manage pre-release number as a build number

### ISSUE THREAD

**Issue by taorepoara** [convco/convco#264]
**Is your feature request related to a problem? Please describe.**

We are using GitLab with fast-forward merges, and we use 3 pre-release channels: `beta`, `rc` and `hotfix`.
We mainly use `beta`, then merge it to `rc`, and then merge it to main to create releases for our projects.

The `hotfix` is made to have a faster workflow to fix production bugs when needed.

The fast-forward merges in GitLab needs the source branch to be rebased from the target branch before the merge.

Our problem:
- we have the main branch corresponding to a release (`1.0.0` for my explanation)
- we have the `rc` that is preparing a new version (pre-release `1.0.0-rc.1`) that is being tested
- meanwhile, we discover a big production bug. We create an `hotfix` version with the fix (`1.0.1-hotfix.1`)
- The fix is tested and merged into the main branch and creates a new release (`1.0.1`)
- We rebase the `rc` branch to include the fix and here comes the problem: the rebase while create a new `rc` pre-release with the same version number as the previous one since the commits do not correspond to previous pre-release (it tries to create a `1.0.0-rc.1` and fails since it already exists).

**Describe the solution you'd like**

IMO, pre-release la number should be an increment from the greatest corresponding pre-release.
In my previous example it should have created a `1.0.0-rc.2` since the `1.0.0-rc.1` already exists.

**Describe alternatives you've considered**

For now, to resolve this problem we have to delete our previous pre-releases to make it work or create one manually.

**Additional context**

**Comment by taorepoara** [comment]
The corresponding tags can be found with the next git command if it can help: `git tag -n v1.1.0-rc.*`

**Comment by taorepoara** [comment]
Yes this behavior works fine.
The problem comes when the rc is preparing a minor version 1.1.0

**Comment by hdevalke** [comment]
Ok I understand the issue and could reproduce it.
The problem is that because of the rebase the tag v1.1.0-rc.1 is not reachable any more.

The question now is how to solve it. Convco version sees tag v1.0.1 as the last one starting from HEAD.
https://github.com/convco/convco/blob/main/src/semver.rs#L52 should probably first check in the versions map https://github.com/convco/convco/blob/main/src/git.rs#L144 the highest version with that prerelease label. 

```
❯ git log --oneline --decorate --all --graph
* 6d8cd28 (HEAD -> rc) feat: release 1.1.0
*   3a9f9f0 (tag: v1.0.1, main) Merge branch 'hotfix'
|\  
| * ae61e0e (tag: v1.0.1-hotfix.1, hotfix) fix: release 1.0.1-hotfix
|/  
| * bb3fed9 (tag: v1.1.0-rc.1) feat: release 1.1.0
|/  
* e4808fe (tag: v1.0.0) feat: release 1.0.0

❯ convco version
1.0.1

❯ convco version --bump --prerelease rc
1.1.0-rc.1

```

**Comment by hdevalke** [comment]
#265 does print now 1.1.0-rc.2 when you run `convco version --bump --prerelease rc` instead of 1.1.0-rc.1.

However, when you run `convco version` it prints 1.0.1 as it cannot know it was rebased.
This was working fine when using merge instead of rebase (and is still working with #265).

```
❯ git log --oneline --decorate --all --graph
*   3e67ef3 (HEAD -> rc) Merge branch 'main' into rc
|\  
| *   3a9f9f0 (tag: v1.0.1, main) Merge branch 'hotfix'
| |\  
| | * ae61e0e (tag: v1.0.1-hotfix.1, hotfix) fix: release 1.0.1-hotfix
| |/  
* / bb3fed9 (tag: v1.1.0-rc.1) feat: release 1.1.0
|/  
* e4808fe (tag: v1.0.0) feat: release 1.0.0

❯ convco version 
1.1.0-rc.1

❯ convco version --bump --prerelease rc
1.1.0-rc.2
```

**Comment by taorepoara** [comment]
This works better but not actually fine...
It gives a different pre-release version at each call.

The find is made but the versions are not sorted so it can give any of the corresponding pre-release: https://github.com/convco/convco/blob/5227823330bb4ec960a34a7e47b98cb9029812fe/src/git.rs#L141
